sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -419,6 +419,14 @@ struct pi_command_list_info_t {
   // TODO: use this for optimizing events in the same command-list, e.g.
   // only have last one visible to the host.
   std::vector<pi_event> EventList{};
+
+  // TODO: remove when the gpu runtime is fixed.
+  // If an error happened at the point of a kernel submission then the
+  // associated fence is not in a valid state. Currently a program hangs if
+  // zeFenceDestroy is called for such fence. This bool variable allows to avoid
+  // the fence destruction if a kernel submission failed.
+  bool IsValidFence{true};
```

Comment:
I have just today submitted a patch for a similar reason, avoid sync of queue that had a failed submission on it: https://github.com/intel/llvm/pull/5524/files#diff-1f19baf7cc76d94bf4a1819c81e7b34292fb3ce215cd50873e8c9236cc73601cR808

Are you sure that it is zeFenceDestroy that is problematic in your case, not the sync of the queue?
And even then, can we maybe avoid destroy of all fences on the non-healthy queue since this is an error condition anyways?

Reply:
oh, I see, sorry, I've just saw your PR. Yes, I am sure that zeFenceDestroy causes a hang in my case (queue sync causes hang as well, but avoiding queue sync is not enough).
I will rework this PR on top of yours.

>And even then, can we maybe avoid destroy of all fences on the non-healthy queue since this is an error condition anyways?
Ok, I will do so.